tools/src/main/java/org/apache/kafka/tools/VerifiableShareConsumer.java

```diff
@@ -0,0 +1,634 @@
+                    ShareGroupAutoOffsetResetStrategy.fromString(this.offsetResetStrategy);
+
+                Properties adminClientProps = new Properties();
+                adminClientProps.put(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, brokerHostandPort);
+
+                Admin adminClient = Admin.create(adminClientProps);
+
+                ConfigResource configResource = new ConfigResource(ConfigResource.Type.GROUP, groupId);
+                Map<ConfigResource, Collection<AlterConfigOp>> alterEntries = new HashMap<>();
+                alterEntries.put(configResource, List.of(new AlterConfigOp(new ConfigEntry(
+                    GroupConfig.SHARE_AUTO_OFFSET_RESET_CONFIG, offsetResetStrategy.type().toString()), AlterConfigOp.OpType.SET)));
+                AlterConfigsOptions alterOptions = new AlterConfigsOptions();
+
+                // Setting the share group auto offset reset strategy
+                adminClient.incrementalAlterConfigs(alterEntries, alterOptions)
+                    .all()
+                    .get(60, TimeUnit.SECONDS);
+
+                printJson(new OffsetResetStrategySet(offsetResetStrategy.type().toString()));
+            }
+
+            consumer.subscribe(Collections.singleton(this.topic));
+            consumer.setAcknowledgementCommitCallback(this);
+            while (!(maxMessages >= 0 && totalAcknowledged >= maxMessages)) {
+                ConsumerRecords<String, String> records = consumer.poll(Duration.ofMillis(5000));
+                if (!records.isEmpty()) {
+                    onRecordsReceived(records);
+
+                    if (acknowledgementMode == AcknowledgementMode.ASYNC) {
+                        consumer.commitAsync();
+                    } else if (acknowledgementMode == AcknowledgementMode.SYNC) {
+                        Map<TopicIdPartition, Optional<KafkaException>> result = consumer.commitSync();
+                        for (Map.Entry<TopicIdPartition, Optional<KafkaException>> resultEntry : result.entrySet()) {
+                            if (resultEntry.getValue().isPresent()) {
+                                log.error("Failed to commit offset synchronously for topic partition: {}", resultEntry.getKey());
+                            }
+                        }
+                    }
+                }
+            }
+        } catch (WakeupException e) {
+            out.println("caught wakeup exception: " + e);
+            // ignore, we are closing
+            log.trace("Caught WakeupException because share consumer is shutdown, ignore and terminate.", e);
+        } catch (Throwable t) {
+            out.println("caught throwable exception: " + t);
+            // Log the error, so it goes to the service log and not stdout
+            log.error("Error during processing, terminating share consumer process: ", t);
+        } finally {
+            out.println("Total records Consumed: " + totalConsumed.toString());
```

Comment:
If we want to write out these pieces of information, I suggest a better way is appropriate. I'd remove these two printlns.